### PR TITLE
Shrink package templates in size and complexity

### PR DESCRIPTION
**Pull request by raman325**
## Breaking change
The entity names have been updated to follow a consistent naming format - `<LOCKNAME>_<DAY OF WEEK (OPTIONAL)>_<ENTITY_NAME>_<CODE_SLOT>`. This means that states will have to be re-input into the new entities and users may have to update their lovelace code

## Proposed change
The package files generated a chain reaction of state changes to entities on any given state change for an input helper or lock state due to the way it was structured. I removed all `binary_sensors` and `sensors` that were just holding logic for another `binary_sensor` or `sensor`, removed the unused `system_ready` stuff from the additional_yaml, and simplified the logic to check whether the pin should be active based on the day, date, time, etc. This should make all of the `keymaster` logic more performant because significantly less states are changing on a given action, and there's less for HomeAssistant to store since we reduced the number of entities created.

Let me know what you think, I think it's still structured in an easily debuggable way but you may feel differently.

BUGFIX:
- `LOCKNAME_active_TEMPLATENUM` binary sensor now handles excluded time range case (it didn't previously)
- There was an automation called `<LOCK NAME> reset` that seemed intended to reset all code slots for the lock. Instead it was calling a manual or conditional notify script, which didn't seem intentional. I have fixed this functionality such that if you flip on reset the lock, it will actually reset the lock

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (which adds functionality)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [x] Code quality improvements to existing code or addition of tests

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue:

### CODE REVIEW

**Comment by raman325** [comment]
Everything here works for my Schlage lock. It would be good to test the template changes for a Kwikset lock to make sure there aren't any lock specific bugs before this gets merged.

**Comment by firstof9** [comment]
i'll give this a try later today with my kwikset lock 👍

**Comment by firstof9** [comment]
Smoke test good.

**Review by firstof9 — APPROVED** [review]

**Comment by raman325** [comment]
FYI it looks like the next version of Home Assistant will require Python 3.8 at a minimum. This will change how we run tests (we would import `AsyncMock` and `patch` from `unittest.mock` instead of importing `patch` from `pytest_homeassistant_custom_component.async_mock`. `pytest_homeassistant_custom_component` has already been updated to reflect the latest changes in home-assistant/core dev branch but we still want to test against Python 3.7 until the change occurs.

TLDR - tests will fail as long as we continue to support 3.7. We can fix the tests but only if we make 3.8 the minimum version for the component

**Comment by firstof9** [comment]
> TLDR - tests will fail as long as we continue to support 3.7. We can fix the tests but only if we make 3.8 the minimum version for the component

On my todo list 👍

**Comment by firstof9** [comment]
Tests are updated.
